docker/init.sh

```diff
@@ -0,0 +1,48 @@
+#! /bin/bash
+
+set -e
+
+# data directory is required for caching the secret key in a file
+if [ "$DATA_DIR" == "" ]; then
+    echo -n "Error: DATA_DIR environment variable not set!"
+    echo "Are you sure you are running this script in a Docker container?"
+    exit 1
+fi
+
+SECRET_KEY_FILE="$DATA_DIR/secret_key"
+
+if [ ! -f "$SECRET_KEY_FILE" ]; then
+    echo ">>> Creating a new secret key file..."
+    install -m 0600 /dev/null "$SECRET_KEY_FILE"
+    SECRET_KEY=$(bundle exec rake secret)
+    echo "$key" > "$SECRET_KEY_FILE"
+    chmod -w "$SECRET_KEY_FILE"
+else
+    SECRET_KEY=$(cat "$SECRET_KEY_FILE")
+fi
+
+export SECRET_KEY
+export RAILS_ENV=production
+
+install -m 0600 /dev/null .my.cnf
+cat > .my.cnf <<ABC
+[client]
+user=$MYSQL_USER
+password=$MYSQL_PASSWORD
+ABC
+
+if [ $(echo "show tables;" | dockerize -wait tcp://$DATABASE_HOST:$DATABASE_PORT -timeout 60s mysql --host $DATABASE_HOST --port $DATABASE_PORT $MYSQL_DATABASE | wc -l) -le 1 ]; then
```

Comment:
`dockerize` is like a launcher for another command that makes sure a host/port combination is reachable, or renders the environment variables into a config file. It guess it should work, if used like `dockerize -wait ... true`.

I just used it twice before because I didn't think of that possibility, and the if clause didn't evaluate every time, and as the command below it requires a database connection, I inserted it there, too. I'll have to test that, and will push a commit.